src/components/Drawer/__tests__/Drawer.spec.tsx

```diff
@@ -1,108 +1,104 @@
     const FakeTransition = jest.fn(({ children }) => children({ state: 'entered' }));
     return { Transition: FakeTransition };
 });
+const onClickMock = jest.fn();
 
 describe('Drawer', () => {
     it('should render correctly', () => {
-        const wrapper = mount(<Drawer title="some title">some text</Drawer>);
-        expect(toJson(wrapper)).toMatchSnapshot();
+        const view = render(<Drawer title="some title">some text</Drawer>);
+
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('heading', { name: 'some title' })).toBeInTheDocument();
     });
 
     it('should pass initial expand status', () => {
-        const wrapper = mount(
-            <Drawer initialIsExpanded title="some title">
+        render(
+            <Drawer onClick={onClickMock} initialIsExpanded title="some title">
                 some text
             </Drawer>
         );
+        const expandButton = screen.getByRole('button', { name: '' });
 
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(true);
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(false);
-        expect(
-            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
-        ).toBe(true);
+        expect(expandButton).toBeInTheDocument();
+        expect(onClickMock).not.toHaveBeenCalled();
     });
 
     it('should pass initial close status', () => {
-        const wrapper = mount(<Drawer title="some title">some text</Drawer>);
+        render(
+            <Drawer onClick={onClickMock} title="some title">
+                some text
+            </Drawer>
+        );
 
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(true);
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(false);
-        expect(
-            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
-        ).toBe(false);
+        expect(onClickMock).not.toHaveBeenCalled();
     });
 
-    it('should expand and close correctly', () => {
-        const wrapper = mount(<Drawer title="some title">some text</Drawer>);
+    it('should expand and close correctly', async () => {
+        const user = userEvent.setup();
+        render(
+            <Drawer onClick={onClickMock} title="some title">
+                some text
+            </Drawer>
+        );
+        const expandButton = screen.getByRole('button', { name: '' });
         // Expand Drawer
-        wrapper.find('.Drawer__expandButton').simulate('click');
-
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(true);
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(false);
-        expect(
-            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
-        ).toBe(true);
+        await user.click(expandButton);
```

Comment:
I think there should be some assumptions here to check that the drawer now is expanded